### ISSUE TITLE
VS editor extremely slow with updating large chunks of code

### ISSUE THREAD

**Issue by cpyrgas** [X-Sharp/XSharpPublic#254]
See attached solution ConsoleApplication27.zip, has one file with long (5,000 lines) but very simple code (a simple assignment repeated). Opening the file happens almost instantly and code is ready for editing with no delay. But if you select all the code, cut or copy it and then paste it in either the same file or another, after pasting the new code becomes visible, but VS stops responding for a long time. In my PC it stops responding for almost one minute! Remember that simply opening the file happens instantly.

[ConsoleApplication27.zip](https://github.com/X-Sharp/XSharpPublic/files/3804507/ConsoleApplication27.zip)

I am pretty sure the same underlying reason causes very big delays also in forms with a lot of controls in them. See also attached solution WindowsFormsApplication46.zip, contains a form with a lot of controls, to to emulate how complex forms are with lots of nested controls and property settings. Opening the form works fast. Same as modifying and save the form, in the first time it works fast as well. 

But now simply open the file Form1.Designer.prg in the ditorand close it immediately. Now modify the form again and save, this time VS becomes again unresponsive for a long time. I assume the Form1.Designer.prg does not really "close" when the user closes it, but remains in memory and the update of the big portion of code that it contains causes the same problem as in the console app above?

[WindowsFormsApplication46.zip](https://github.com/X-Sharp/XSharpPublic/files/3804508/WindowsFormsApplication46.zip)

**Comment by JamalAPI** [comment]
Hi Chris,

Tested in VS 2019 Pro. 

Here is what I got after testing the ConsoleApplication27 copy/paste.  It froze for 24 seconds. See yellow line on top of image.

It seems diagnostics or analysis are going on in the background.

![2019-11-04_9-59-24](https://user-images.githubusercontent.com/35046731/68132400-9eb66000-feec-11e9-911a-3b6d6b509a6d.png)
.
BTW, in a C# console app, copy/paste same code is instantaneous.
